src/main/java/org/folio/service/titles/TitlesService.java

```diff
@@ -130,16 +158,212 @@ private List<String> getNonPackageLineIds(List<CompositePoLine> compositePoLines
+        .map(v -> holder))
+      .compose(holder -> deleteTitle(holder.getTitleId(), requestContext));
+  }
+
+  private Future<Void> processHoldingItemPieces(TitleHolder holder, String deleteHoldings, RequestContext requestContext) {
+    List<String> holdingIdsToDelete = holder.getHoldingIdsToDelete();
+
+    if (CollectionUtils.isEmpty(holdingIdsToDelete)) {
+      return Future.succeededFuture();
+    }
+
+    if (StringUtils.isEmpty(deleteHoldings)) {
+      log.info("processHoldings:: Holdings in poLine '{}' will not be deleted", holder.getPoLine().getId());
+      var param = new Parameter().withKey("holdingIds").withValue(holdingIdsToDelete.toString());
+      var error = EXISTING_HOLDINGS_FOR_DELETE_CONFIRMATION.toError().withParameters(List.of(param));
+      throw new HttpException(BAD_REQUEST, error);
+    }
+
+    if (!"true".equalsIgnoreCase(deleteHoldings) && !"false".equalsIgnoreCase(deleteHoldings)) {
+      throw new IllegalArgumentException("deleteHolding must be either 'true' or 'false'");
+    }
+
+    return Boolean.parseBoolean(deleteHoldings)
+      ? deleteHoldingItemPieces(holder, requestContext)
+      : Future.succeededFuture();
+  }
+
+  private Future<Map<String, List<String>>> getHoldingsToDeleteWithTenants(TitleHolder holder, RequestContext requestContext) {
+    return getHoldingIdsGroupedByTenantId(holder.getTitle(), holder.getPoLine(), requestContext)
+      .compose(holdingIdsGroupedByTenant -> extractDeletableHoldings(holder, holdingIdsGroupedByTenant, requestContext));
+  }
+
+  private Future<Map<String, List<String>>> getHoldingIdsGroupedByTenantId(Title title, PoLine poLine, RequestContext requestContext) {
+    return pieceStorageService.getPiecesByLineIdAndTitleId(poLine.getId(), title.getId(), requestContext)
+      .map(pieces -> pieces.stream()
+        .filter(piece -> StringUtils.isNotEmpty(piece.getHoldingId()))
+        .collect(groupingBy(Piece::getReceivingTenantId, mapping(Piece::getHoldingId, toList())))
+      );
+  }
+
+  private Future<Map<String, List<String>>> extractDeletableHoldings(TitleHolder holder, Map<String, List<String>> holdingIdsGroupByTenant,
+                                                                     RequestContext requestContext) {
+    if (holdingIdsGroupByTenant.isEmpty()) {
+      return Future.succeededFuture(Map.of());
+    }
+
+    List<Future<AbstractMap.SimpleEntry<String, String>>> deletableHoldingsFuture = new ArrayList<>();
+    holdingIdsGroupByTenant.forEach((tenantId, holdingIds) -> {
+      var tenantContext = holder.isCentralEnabled()
+        ? createContextWithNewTenantId(requestContext, tenantId)
+        : requestContext;
+
+      holdingIds.forEach(holdingId -> deletableHoldingsFuture.add(
+        inventoryItemManager.getItemsByHoldingId(holdingId, tenantContext)
+          .map(items -> items.stream()
+            .anyMatch(item -> holder.getPoLineId().equals(item.getString("purchaseOrderLineIdentifier")))
```

Comment:
There is constant in InventoryItemManager ITEM_PURCHASE_ORDER_LINE_IDENTIFIER that we can use here